flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/table/DynamoDbDynamicSinkFactory.java

```diff
@@ -58,6 +58,17 @@ public DynamicTableSink createDynamicTableSink(Context context) {
                         .setDynamoDbClientProperties(
                                 dynamoDbConfiguration.getSinkClientProperties());
 
+        if (catalogTable.getResolvedSchema().getPrimaryKey().isPresent()) {
```

Comment:
IN case of DynamoDB contents of PrimaryKey and Partition Key should be the same to ensure semantic correctness of various operations. 

To simplify the usage, we should aim to use just PrimaryKey property, as it's a fairly intuitive and well understood concept for DDB users. In the meantime, we should probably aim to start deprecation process of the `partition by` mechanism. 

The first step could be 
- if PK is set, but `partition by` is not, use PK as `partition by` 
- document that `partition by` is on deprecation path
- log warnings when `partition by` is being used

Reply:
+1
Also, I'd add a check that, if `PARTITION BY` and `PRIMARY KEY` are both specified, they must be identical

Reply:
@dzikosc - Isn't there a potential use case wherein a user of the existing `PARTITION BY` functionality of the sink wants to use it to deduplicate data by something other than the actual Primary Key of the DynamoDB table?   For example, the DynamoDB table has:

```
user_id  Partition Key 
order_id Sort Key 
line_number 
price
```

Today, the user of the connector could specify a `PARTITION BY  user_id, order_id, line_number` to deduplicate incoming data - Maybe this is a contrived case, but just pointing out that we may now be blocking a use case that was previously supported.   Thoughts?  

Either way, could we keep the deprecation of `PARTITION BY` to a separate PR to reduce the scope of this one?